### PR TITLE
prevent <ctrl>-<enter> handler from <space> from toggling checkboxes

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR...

Fixes: #6017

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

manually tested the register template form.

#### testing each of the changes one by one, as the solution seems too generic

- [x] fixed: ui/src/components/view/ResourceLimitTab.vue
- [ ]	modified:   ui/src/views/AutogenView.vue
- [x] checked:   ui/src/views/auth/Login.vue
- [x] c:   ui/src/views/compute/AttachIso.vue
- [ ]	modified:   ui/src/views/compute/ChangeAffinity.vue
- [ ]	modified:   ui/src/views/compute/CreateKubernetesCluster.vue
- [ ]	modified:   ui/src/views/compute/CreateSSHKeyPair.vue
- [ ]	modified:   ui/src/views/compute/CreateSnapshotWizard.vue
- [ ]	modified:   ui/src/views/compute/DeployVM.vue
- [ ]	modified:   ui/src/views/compute/DestroyVM.vue
- [ ]	modified:   ui/src/views/compute/EditVM.vue
- [ ]	modified:   ui/src/views/compute/ScaleKubernetesCluster.vue
- [ ]	modified:   ui/src/views/compute/ScaleVM.vue
- [ ]	modified:   ui/src/views/compute/StartVirtualMachine.vue
- [ ]	modified:   ui/src/views/compute/UpgradeKubernetesCluster.vue
- [ ]	modified:   ui/src/views/compute/backup/FormSchedule.vue
- [x] f: ui/src/views/iam/AddAccount.vue
- [ ]	modified:   ui/src/views/iam/AddLdapAccount.vue
- [ ]	modified:   ui/src/views/iam/AddUser.vue
- [ ]	modified:   ui/src/views/iam/ChangeUserPassword.vue
- [ ]	modified:   ui/src/views/iam/ConfigureSamlSsoAuth.vue
- [ ]	modified:   ui/src/views/iam/CreateRole.vue
- [ ]	modified:   ui/src/views/iam/DomainActionForm.vue
- [ ]	modified:   ui/src/views/iam/EditUser.vue
- [ ]	modified:   ui/src/views/iam/ImportRole.vue
- [ ]	modified:   ui/src/views/image/AddKubernetesSupportedVersion.vue
- [x] f: ui/src/views/image/RegisterOrUploadIso.vue
- [x] f: ui/src/views/image/RegisterOrUploadTemplate.vue
- [ ]	modified:   ui/src/views/image/UpdateKubernetesSupportedVersion.vue
- [ ]	modified:   ui/src/views/image/UpdateTemplate.vue
- [ ]	modified:   ui/src/views/infra/AddPrimaryStorage.vue
- [ ]	modified:   ui/src/views/infra/AddSecondaryStorage.vue
- [ ]	modified:   ui/src/views/infra/MigrateData.vue
- [ ]	modified:   ui/src/views/infra/PodAdd.vue
- [ ]	modified:   ui/src/views/infra/network/DedicatedVLANTab.vue
- [ ]	modified:   ui/src/views/infra/network/EditTrafficLabel.vue
- [ ]	modified:   ui/src/views/infra/network/ServiceProvidersTab.vue
- [ ]	modified:   ui/src/views/infra/network/providers/AddF5LoadBalancer.vue
- [ ]	modified:   ui/src/views/infra/network/providers/AddNetscalerLoadBalancer.vue
- [ ]	modified:   ui/src/views/infra/network/providers/AddNiciraNvpDevice.vue
- [ ]	modified:   ui/src/views/infra/network/providers/AddPaloAltoFirewall.vue
- [ ]	modified:   ui/src/views/infra/network/providers/AddSrxFirewall.vue
- [ ]	modified:   ui/src/views/infra/zone/AdvancedGuestTrafficForm.vue
- [ ]	modified:   ui/src/views/infra/zone/IpAddressRangeForm.vue
- [ ]	modified:   ui/src/views/infra/zone/StaticInputsForm.vue
- [ ]	modified:   ui/src/views/infra/zone/ZoneWizardPhysicalNetworkSetupStep.vue
- [ ]	modified:   ui/src/views/infra/zone/ZoneWizardZoneDetailsStep.vue
- [ ]	modified:   ui/src/views/infra/zone/ZoneWizardZoneTypeStep.vue
- [ ]	modified:   ui/src/views/network/CreateIsolatedNetworkForm.vue
- [ ]	modified:   ui/src/views/network/CreateL2NetworkForm.vue
- [ ]	modified:   ui/src/views/network/CreateSharedNetworkForm.vue
- [ ]	modified:   ui/src/views/network/CreateVlanIpRange.vue
- [ ]	modified:   ui/src/views/network/CreateVpc.vue
- [ ]	modified:   ui/src/views/network/CreateVpnCustomerGateway.vue
- [ ]	modified:   ui/src/views/network/InternalLBAssignVmForm.vue
- [ ]	modified:   ui/src/views/network/UpdateNetwork.vue
- [ ]	modified:   ui/src/views/offering/AddComputeOffering.vue
- [ ]	modified:   ui/src/views/offering/AddDiskOffering.vue
- [ ]	modified:   ui/src/views/offering/AddNetworkOffering.vue
- [ ]	modified:   ui/src/views/offering/AddVpcOffering.vue
- [ ]	modified:   ui/src/views/offering/ImportBackupOffering.vue
- [ ]	modified:   ui/src/views/offering/UpdateOfferingAccess.vue
- [ ]	modified:   ui/src/views/plugins/quota/EmailTemplateDetails.vue
- [ ]	modified:   ui/src/views/project/InvitationTokenTemplate.vue
- [ ]	modified:   ui/src/views/storage/AttachVolume.vue
- [ ]	modified:   ui/src/views/storage/CreateSnapshotFromVMSnapshot.vue
- [ ]	modified:   ui/src/views/storage/CreateVolume.vue
- [ ]	modified:   ui/src/views/storage/FormSchedule.vue
- [ ]	modified:   ui/src/views/storage/ResizeVolume.vue
- [ ]	modified:   ui/src/views/storage/RestoreAttachBackupVolume.vue
- [ ]	modified:   ui/src/views/storage/TakeSnapshot.vue
- [ ]	modified:   ui/src/views/storage/UploadLocalVolume.vue

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan ui

**Comment by DaanHoogland** [comment]
@blueorangutan ui

**Comment by DaanHoogland** [comment]
@utchoang can you advice if there is a better generic solution to handling keyboard navigation?

**Comment by DaanHoogland** [comment]
I reverted the generic change and limited to the register template form. I'm sure more of these issues will come up.

**Comment by DaanHoogland** [comment]
@blueorangutan ui

File: ui/src/views/image/RegisterOrUploadTemplate.vue [hunk 1]
```diff
@@ -16,15 +16,16 @@
...
+    class="form-layout"
+    @keyup.ctrl.enter="handleSubmit">
     <span v-if="uploadPercentage > 0">
       <a-icon type="loading" />
       {{ $t('message.upload.file.processing') }}
       <a-progress :percent="uploadPercentage" />
     </span>
     <a-spin :spinning="loading" v-else>
       <a-form
-        v-ctrl-enter="handleSubmit"
```
Comment (sureshanaparti) [inline ui/src/views/image/RegisterOrUploadTemplate.vue:27]: @DaanHoogland Should we remove here as well ?

https://github.com/apache/cloudstack/blob/1a47cc3c954ded8db7afdc8a0154929ae54c6c48/ui/src/views/image/RegisterOrUploadIso.vue#L29

Comment (DaanHoogland) [reply ui/src/views/image/RegisterOrUploadTemplate.vue:27]: yes, this is the one obfuscating the `space` event. that's why it was moved to the `div` element

Comment (sureshanaparti) [reply ui/src/views/image/RegisterOrUploadTemplate.vue:27]: ok, is the same change required in _RegisterOrUploadIso.vue_

Comment (DaanHoogland) [reply ui/src/views/image/RegisterOrUploadTemplate.vue:27]: I am working on that, but the RegisterOrUploadIso doesn't use `a-check`s but `a-switch`s, which work differently. these will not accept the  `space` event. I think we should merge like this and handle the rest when we encounter it.

**Review by Pearl1594 — APPROVED** [review]
LGTM.
There are a few other forms where we have checkboxes that are effected which include:
- Create Network Offering
- Create VPC offering
- Assign LB rule to VM
However, we can address these in 4.17 too.

**Comment by DaanHoogland** [comment]
> LGTM. There are a few other forms where we have checkboxes that are effected which include:
> 
>     * Create Network Offering
> 
>     * Create VPC offering
> 
>     * Assign LB rule to VM
>       However, we can address these in 4.17 too.

tnx @Pearl, added.

**Comment by DaanHoogland** [comment]
@blueorangutan ui

**Comment by sureshanaparti** [comment]
Manually tested the forms for 'Register template', 'Create Network Offering', 'Create VPC offering' forms. LGTM

**Review by sureshanaparti — APPROVED** [review]

**Review by nvazquez — APPROVED** [review]
LGTM

**Pull request merged**